### DISCUSSION TITLE
ssh: connect to host ssh.github.com port 22: Operation timed out

### DISCUSSION THREAD

**Discussion by huyvohcmc** [community/community#54637] [Repositories]
### Select Topic Area

General

### Body

I already set up GitHub SSH but for the past few days none of my git commands seem to work: git pull, git push, git fetch, etc all got timed out.
```
> ssh -T git@ssh.github.com
ssh: connect to host ssh.github.com port 22: Operation timed out

> git pull
ssh: connect to host github.com port 22: Operation timed out
fatal: Could not read from remote repository.

Please make sure you have the correct access rights and the repository exists.
```

My .ssh/config:
```
Host github.com
  AddKeysToAgent yes
  IdentityFile ~/.ssh/id_ed25519
```

I asked my internet provider, and they confirmed they don't block GitHub SSH access. I am now unable to access GitHub without using VPN.

Is there anything else that I can do to troubleshoot this issue?

**Comment by airtower-luna** [comment]
Your problem sounds very similar to the one here:

* #54558

If a VPN works you can rule out a misconfigured firewall on your own computer, but somewhere else along the path is still possible. As I described there you can use `traceroute` (might work slight differently depending on the OS) to figure out *where* packets get lost, and you could try SSH over the HTTPS port.

If [SSH over the HTTPS port](https://github.com/orgs/community/discussions/54558#discussioncomment-5823514) works the SSH packets on the regular port get eaten somewhere, and you might need to talk to your ISP again: Even if they don't intentionally block SSH there might be a bug, or an issue with one of the networks they connect to in turn.

**Reply by huyvohcmc** [reply]
Port 443 doesn't work for me either

**Reply by airtower-luna** [reply]
Port 443 on which host? It should only work on `ssh.github.com`.

**Reply by huyvohcmc** [reply]
Yep, I tried it on `ssh.github.com`. Luckily the issue was sorted out by my ISP today.

**Comment by jcmendez26** [comment]
========= EN EL CASO DE MOSTRAR EL SIGUIENTE ERROR POR FALLO DE LA COMPROBACION DE CONEXION====================
ssh: connect to host github.com port 22: Connection timed out

CREAR EL ARCHIVO "config" FORZOSAMENTE dentro de la carpeta ".ssh" que se crea en el directorio del usuario de la PC
'/c/Users/Nomb_UserPC/.ssh/config'

y agregar las siguientes lineas dentro del archivo "config":

Host github.com
AddKeysToAgent yes
#revisar la ruta y nombre del archivo de la clave SSH creada, deben coincidir (pueden borrar esta línea)
IdentityFile ~/.ssh/nomb_arch_claveSSH
Hostname ssh.github.com
Port 443

Probado en una PC con Windows 11 Pro

**Reply by aaron-ev** [reply]
Thank you!, it worked for me.

**Reply by tearsoflebesgue** [reply]
Great！

**Reply by OliverLiy** [reply]
thank you!

**Comment by saidbakr** [comment]
In some circumstances **Your Firewall** settings is the cause of that issue. I mean by firewall both those on your device and on your network gateway i.e . your router for example.
In my case, My router's firewall was the responsible to this issue, I just changed its firewall settings from *Strict* to *Standard* and then everything works fine.
![image](https://github.com/community/community/assets/467554/8f46f100-7d55-4f11-8b5c-9eab1b31abae)

**Comment by Azieyati** [comment]
Try this:
`ssh -T git@ssh.github.com -p 443`

**Reply by fernandacorreiasoares** [reply]
Nice! 
tnks a lot!

**Reply by jaeyson** [reply]
This helped, thanks alot!!!

**Reply by treedenPP** [reply]
Thanks! It works!

**Reply by r7vera** [reply]
Thanks

**Reply by MayHuo** [reply]
This helped, thanks a lot!!!

**Comment by dotrinh-USE-Inc** [comment]
I disable VPN, it works

**Comment by JamesV-1** [comment]
I changed wifi networks and that fixed the issue for me

**Reply by Vramiro7** [reply]
I was attempting to create a new repo at work and it was so tedious until I thought of firewall. So I switched to my phone's hotspot and BOOM! was able to push right through
